src/datasources/balances-api/zerion-balances-api.service.ts

```diff
@@ -210,6 +210,9 @@ export class ZerionBalancesApi implements IBalancesApi {
           zerionCollectibles.data,
         );
       } catch (error) {
+        if (error instanceof ZodError) {
```

Comment:
The same as the previous comment, Wouldn't it be better to move it to the factory as that's the class that handles error related logic?

Reply:
See https://github.com/safe-global/safe-client-gateway/pull/2166#discussion_r1869790236.